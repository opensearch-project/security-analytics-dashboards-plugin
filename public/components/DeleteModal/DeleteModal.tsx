/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { ChangeEvent, Component } from 'react';
import {
  EuiConfirmModal,
  EuiCompressedFieldText,
  EuiForm,
  EuiCompressedFormRow,
  EuiOverlayMask,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';

interface DeleteModalProps {
  additionalWarning?: string;
  closeDeleteModal: (event?: any) => void;
  confirmation?: boolean;
  ids: string;
  onClickDelete: (event?: any) => void;
  type: string;
  confirmButtonText?: string;
}

interface DeleteModalState {
  confirmDeleteText: string;
}

export const DEFAULT_DELETION_TEXT = 'delete';

export default class DeleteModal extends Component<DeleteModalProps, DeleteModalState> {
  constructor(props: DeleteModalProps) {
    super(props);
    const { confirmation } = props;
    this.state = {
      confirmDeleteText: confirmation ? '' : DEFAULT_DELETION_TEXT,
    };
  }

  onChange = (e: ChangeEvent<HTMLInputElement>): void => {
    this.setState({ confirmDeleteText: e.target.value });
  };

  render() {
    const {
      type,
      ids,
      closeDeleteModal,
      onClickDelete,
      additionalWarning,
      confirmation,
      confirmButtonText,
    } = this.props;
    const { confirmDeleteText } = this.state;

    return (
      <EuiOverlayMask>
        <EuiConfirmModal
          title={<EuiText size="s"><h2>`Delete ${type}`</h2></EuiText>}
          onCancel={closeDeleteModal}
          onConfirm={() => {
            onClickDelete();
            closeDeleteModal();
          }}
          cancelButtonText={'Cancel'}
          confirmButtonText={confirmButtonText ?? `Delete ${type}`}
          buttonColor={'danger'}
          defaultFocusedButton="confirm"
          confirmButtonDisabled={confirmDeleteText != DEFAULT_DELETION_TEXT}
        >
          <EuiForm>
            <p>
              Delete "<strong>{ids}</strong>" permanently? {additionalWarning}
            </p>
            <EuiSpacer size="s" />
            {!!confirmation && (
              <EuiCompressedFormRow helpText={`To confirm deletion, type "${DEFAULT_DELETION_TEXT}".`}>
                <EuiCompressedFieldText
                  value={confirmDeleteText}
                  placeholder={DEFAULT_DELETION_TEXT}
                  onChange={this.onChange}
                  data-test-subj={'deleteTextField'}
                />
              </EuiCompressedFormRow>
            )}
          </EuiForm>
        </EuiConfirmModal>
      </EuiOverlayMask>
    );
  }
}
