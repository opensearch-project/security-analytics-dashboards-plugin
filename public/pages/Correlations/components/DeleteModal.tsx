/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiConfirmModal, EuiText } from '@elastic/eui';
import React from 'react';

export interface DeleteRuleModalProps {
  title: string;
  onCancel: () => void;
  onConfirm: () => void;
}

export const DeleteCorrelationRuleModal: React.FC<DeleteRuleModalProps> = ({
  title,
  onCancel,
  onConfirm,
}) => {
  return (
    <EuiConfirmModal
      title={<EuiText size="s"><h2>Delete {title}?</h2></EuiText>}
      onCancel={onCancel}
      onConfirm={onConfirm}
      cancelButtonText="Cancel"
      confirmButtonText="Delete"
      buttonColor="danger"
      defaultFocusedButton="confirm"
    >
      <EuiText size="s">Delete the correlation rule permanently? This action cannot be undone.</EuiText>
    </EuiConfirmModal>
  );
};
