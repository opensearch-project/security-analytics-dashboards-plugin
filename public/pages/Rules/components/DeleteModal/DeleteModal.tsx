/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiConfirmModal } from '@elastic/eui';
import React from 'react';

export interface DeleteRuleModalProps {
  title: string;
  onCancel: () => void;
  onConfirm: () => void;
}

export const DeleteRuleModal: React.FC<DeleteRuleModalProps> = ({ title, onCancel, onConfirm }) => {
  return (
    <EuiConfirmModal
      title={`Delete ${title}?`}
      onCancel={onCancel}
      onConfirm={onConfirm}
      cancelButtonText="Cancel"
      confirmButtonText="Delete"
      buttonColor="danger"
      defaultFocusedButton="confirm"
    >
      <p>Delete the rule permanently? This action cannot be undone.</p>
    </EuiConfirmModal>
  );
};
