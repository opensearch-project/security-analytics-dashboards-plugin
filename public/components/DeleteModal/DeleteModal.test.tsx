/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import DeleteModal from './DeleteModal';

describe('<DeleteModal /> spec', () => {
  it('renders the component', () => {
    const tree = render(
      <DeleteModal
        closeDeleteModal={() => jest.fn()}
        ids={'some ids'}
        onClickDelete={() => jest.fn()}
        type={'some type'}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
