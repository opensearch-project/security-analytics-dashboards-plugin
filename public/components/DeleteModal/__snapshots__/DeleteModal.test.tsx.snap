// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteModal /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="euiBody-hasOverlayMask"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      aria-hidden="true"
      class="euiOverlayMask euiOverlayMask--aboveHeader"
      data-aria-hidden="true"
    />
    <div
      class="euiOverlayMask euiOverlayMask--aboveHeader"
    >
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="euiModal euiModal--maxWidth-default euiModal--confirmation"
          tabindex="0"
        >
          <button
            aria-label="Closes this modal window"
            class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
            type="button"
          >
            EuiIconMock
          </button>
          <div
            class="euiModal__flex"
          >
            <div
              class="euiModalHeader"
            >
              <div
                class="euiModalHeader__title"
                data-test-subj="confirmModalTitleText"
              >
                <div
                  class="euiText euiText--small"
                >
                  <h2>
                    \`Delete $
                    some type
                    \`
                  </h2>
                </div>
              </div>
            </div>
            <div
              class="euiModalBody"
            >
              <div
                class="euiModalBody__overflow"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="confirmModalBodyText"
                >
                  <div
                    class="euiForm"
                  >
                    <p>
                      Delete "
                      <strong>
                        some ids
                      </strong>
                      " permanently? 
                    </p>
                    <div
                      class="euiSpacer euiSpacer--s"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiModalFooter"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--primary"
                data-test-subj="confirmModalCancelButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonEmpty__content"
                >
                  <span
                    class="euiButtonEmpty__text"
                  >
                    Cancel
                  </span>
                </span>
              </button>
              <button
                class="euiButton euiButton--danger euiButton--fill"
                data-test-subj="confirmModalConfirmButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Delete some type
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
